blueprint/core/blueprint_ViewManager.h

```diff
@@ -0,0 +1,87 @@
+
+namespace blueprint
+{
+    //TODO: Document
+    class ViewManager : public juce::ComponentListener
+    {
+    public:
+        //==============================================================================
+        // We allow registering arbitrary view types with the React context by way of
+        // a "ViewFactory" here which is a user-defined function that produces a View
+        // and a corresponding ShadowView.
+        typedef std::pair<std::unique_ptr<View>, std::unique_ptr<ShadowView>> ViewPair;
+        typedef std::function<ViewPair()> ViewFactory;
+
+        //==============================================================================
+
+        //TODO: Document
+        explicit ViewManager(View* rootView);
+        ~ViewManager();
+
+        /** Registers a new dynamic view type and its associated factory. */
+        void registerViewType(const juce::String& typeId, ViewFactory f);
+
+        /** Creates a new view instance and registers it with the view table. */
+        ViewId createViewInstance(const juce::String& viewType);
+
+        /** Creates a new text view instance and registers it with the view table. */
+        ViewId createTextViewInstance(const juce::String& value);
+
+        //TODO: Document
+        void setViewProperty(ViewId viewId, const juce::Identifier& name, const juce::var& value);
+
+        //TODO: Document
+        void setRawTextValue(ViewId viewId, const juce::String& value);
+
+        //TODO: Document
+        void addChild(ViewId parentId, ViewId childId, int index = -1);
+
+        //TODO: Document
+        void removeChild (ViewId parentId, ViewId childId);
+
+        //==============================================================================
+        // juce::ComponentListener
+        void componentMovedOrResized (juce::Component& component, bool wasMoved, bool wasResized) override;
+
+        //==============================================================================
+    private:
+        void enumerateChildViewIds (std::vector<ViewId>& ids, View* v);
+
+        /** Recursively computes the shadow tree layout on the root ShadowView, then traverses the tree
+            flushing new layout bounds to the associated view components.
+         */
+        void performRootShadowTreeLayout();
+
+        /** Returns a pointer pair to the view associated to the given id. */
+        //TODO: Private for now as only used internally but any desire for this to be made public?
+        //      so classes like ReactApplicationRoot can fetch a View?
+        std::pair<View*, ShadowView*> getViewHandle (ViewId viewId);
+
+        /** Walks the view table, returning the first view with a `refId`
+         *  whose value equals the provided id.
+         */
+         //TODO: Currently unused. Keep?
+        View* getViewByRefId (const juce::Identifier& refId);
+
+        std::unique_ptr<ShadowView>  rootShadowView;
+
+        std::map<ViewId, std::unique_ptr<View>> viewTable;
```

Comment:
Not sure how you feel about style @nick-thompson but be nice to align all these member var names

Reply:
Sure! What do you suggest?

Reply:
@nick-thompson cleaned up in https://github.com/nick-thompson/blueprint/pull/46/commits/9aba516339d06f6ff9a3d03a3bb5ce9bba592c7d